net/lp_rpc.proto

```diff
@@ -368,6 +376,37 @@ message NotifySegment {
     reserved 33; // Formerly "repeated VideoProfile fullProfiles"
 }
 
+// Sent by the aiworker to register itself to the orchestrator.
+message RegisterAIWorkerRequest {
+
+    // Shared secret for auth
+    string secret = 1;
+
+    // AIWorker capacity
+    int64 capacity = 2;
+
+    // AIWorker capabilities
+    Capabilities capabilities = 3;
+}
+
+// Sent by the orchestrator to the aiworker
+message NotifyAIJob {
+    // URL of the input to use.
+    string url      = 1;
+
+    // pipeline to use for the job
+    string pipeline = 2;
+
+    // model to process job with
+    string modelID = 3;
```

Comment:
Nico's version uses a enum for pipeline that will need to be updated with each new pipeline and a data field that i think encodes the modelID into it. The data field also carries all the binary data for the request I believe which puts a cap on the input file size at 4mb (gRPC server size limit)

Reply:
I pull out modelID so the ai-worker can reserve the GPU before type checking the `requestData`.  modelID is included in all requests though. Can I get it without type checking? Maybe just convert it to json?

Link to request structs created with codegen https://github.com/livepeer/ai-worker/blob/ddda616fe0e266a60ab92f6263cefb1d7598e6c4/worker/runner.gen.go#L45

Reply:
I see, so you're using this `modelID` here:
https://github.com/livepeer/go-livepeer/blob/139f0347983febbf3c522e14f79498273255a006/server/ai_worker.go#L161

Before parsing `requestData` here:
https://github.com/livepeer/go-livepeer/blob/139f0347983febbf3c522e14f79498273255a006/server/ai_worker.go#L188

IMO parsing request data should be fast, so not really an optimization. Actually it may even work worse, because if the requestData is not parsed correctly, we don't want to allocate GPU.

All in all, if you prefer to have `modelID` in the proto, I'm ok with that as long as we're sure that `modelID` makes sense for each request (which I think is true). However, if `url` is not always present then I'd definitely include it in `requestData`.

Reply:
I have updated to add `AIJobData` to proto file similar to `SegData` so that the specifics of the request are grouped together in the `NotifyAIJob` message sent to the ai-worker.  I had to update a number of the tests to account for this and confirmed all are passing.

The Orchestrator verifies the request received with the open api validation so it should be good once it gets sent to the ai-worker (see route registrations in server/ai_http.go). The type checking at the ai-worker level is mostly to correctly parse the request to the auto generated types from ai-worker repo and confirm nothing went weird in transit.